evm/ecc/backends/coincurve.py

```diff
@@ -0,0 +1,77 @@
+from .base import BaseECCBackend
+
+from coincurve.keys import (
```

Comment:
I would request that all `coincurve` imports be positioned within the `CoinCurveECCBackend` class itself so that this module can be properly imported even if `coincurve` isn't available.  Then, within the `__init__` method of the `CoinCurveECCBackend` class, do something like the following to produce a nice friendly import error if the library isn't available.

```python
def __init__(....):
    try:
        import coincurve
    except ImportError:
        raise ImportError("The CoinCurveECCBackend requires the coincurve library which is not available for import.")
    ...
```